virttest/utils_net.py

```diff
@@ -3066,50 +3066,65 @@ def parse_arp():
     return ret
 
 
-def verify_ip_address_ownership(ip, macs, timeout=60.0, devs=None):
+def verify_ip_address_ownership(ip, macs, timeout=20.0, devs=None):
     """
-    Use arping and the ARP cache to make sure a given IP address belongs to one
-    of the given MAC addresses.
+    Make sure a given IP address belongs to one of the given
+    MAC addresses.
 
-    :param ip: An IP address.
+    :param ip: The IP address to be verified.
     :param macs: A list or tuple of MAC addresses.
-    :param devs: A set of network interfaces to check on. If is absent then use
-                 route table to get a name for possible network interfaces.
+    :param devs: A set of network interfaces to check on. If is absent
+                 then use route table to get a name for possible
+                 network interfaces.
     :return: True if ip is assigned to a MAC address in macs.
     """
-    def __arping(regex, arping_cmd, ip):
+    def __arping(ip, macs, dev, timeout):
         # Compile a regex that matches the given IP address and any of the
         # given MAC addresses from arping output
         ip_map = parse_arp()
         for mac in macs:
             if ip_map.get(mac) == ip:
                 return True
+
+        mac_regex = "|".join("(%s)" % mac for mac in macs)
+        regex = re.compile(r"\b%s\b.*\b(%s)\b" % (ip, mac_regex), re.I)
+        arping_bin = utils_path.find_command("arping")
+        arping_cmd = "%s -f -c3 -w%d -I %s %s" % (arping_bin, int(timeout),
+                                                  dev, ip)
         s, o = commands.getstatusoutput(arping_cmd)
         if s != 0:
             return False
         return bool(regex.search(o))
 
+    def __verify_neigh(ip, macs, dev, timeout):
```

Comment:
`timeout` is not used here.

Reply:
It's true, but for being easily used at L3126, it requires the dummy one.